### PR TITLE
Implement `buffer_sub_data`

### PR DESCRIPTION
**Pull request by kyren**
May technically require a version bump of web-sys?

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -208,6 +208,8 @@ pub trait Context {
 
     unsafe fn buffer_data_u8_slice(&self, target: u32, data: &[u8], usage: u32);
 
+    unsafe fn buffer_sub_data(&self, target: u32, offset: i32, src_data: &[u8]);
```
Comment (grovesNL) [inline src/lib.rs]: Could we match the naming of `buffer_data_u8_slice`, like `buffer_sub_data_u8_slice`?

Comment (kyren) [reply src/lib.rs]: Sure thing, I didn't know if you wanted the same naming convention even though there's only the one method.

**Comment by kyren** [comment]
> What changed in web-sys? I think it's fine to bump the web-sys version

I think it was one of the many `&mut [u8]` to `&[u8]` changes that this PR is relying on, it worked *after* a call to `cargo update`.

**Comment by grovesNL** [comment]
Alright, I think we should probably bump web-sys in that case. We can fix the other `&mut`s later

**Pull request merged**